File: backend/auth/src/main/java/com/bombi/auth/presentation/dto/CommonResponseDto.java
package com.bombi.auth.presentation.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL) // 필드가 null이면 직렬화에서 제외
public class CommonResponseDto<T> {
    private String status;
    private String message;

    private T data; // 요청별 응답 데이터

    public CommonResponseDto(String status, String message) {
        this.status = status;
        this.message = message;
    }
}

File: backend/llm/app/dto/common_response_dto.py
from typing import TypeVar, Generic, Optional
from pydantic import BaseModel  # ✅ 최신 pydantic 적용

T = TypeVar("T")


class CommonResponseDto(BaseModel, Generic[T]):
    status: str  # ✅ 필수 필드 (누락 시 오류 발생)
    message: str
    data: Optional[T] = None

